### PR TITLE
Decorate

### PR DESCRIPTION
**Pull request by fherrerao**
# Hi There 💯 
### Features implemented in this branch:

- Created Nameable class and correct_name method that raises a NotImplementedError.
- Created Decorator class that inherits from Nameable and returns the name attribute.
- Created CapitalizeDecorator and TrimmerDecorator classes that inherit from Decorator and implement a method correct_name that capitalizes the output of @nameable.correct_name.

Please let me know if I should make changes.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (TedLivist) [inline capitalize_decorator.rb]: - [x] You are required to return a capitalized word (eg. 'Maximilianus'), not an uppercase (eg 'MAXIMILIANUS'). Kindly fix that.

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 9 ? "#{@nameable.correct_name[0..8]} ..." : @nameable.correct_name
```
Comment (TedLivist) [inline trimmer_decorator.rb]: - [x] This Trimmer returns only 9 characters where you are expected to return 10 characters. Kindly do that.

  You can change `[0..8]` to `[0..9]` to resolve this.

- [x] Note that you simply expected to return the trimmed correct_name, not a string particularly. Kindly remove the string in the conditional

File: nameable.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Nameable
+  def initialize; end
```
Comment (TedLivist) [inline nameable.rb]: - [x] **[OPTIONAL]** This `initialize` method is unused and so unnecessary. I would recommend you remove it

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @fherrerao,

Good job so far!
✔️ Good documentation
✔️ Creation of Base class
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 2 above)
Comment (fherrerao) [reply trimmer_decorator.rb]: Thank you for your feedback, I just fixed the mistakes.

File: trimmer_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length >= 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
```
Comment (medaminedev66) [inline trimmer_decorator.rb]: - [x] The length of the name in the condition should be greater than 10, not greater or equal to 10. Please remove the symbol `=`. Remember, if the name length is longer than 10 then it should trim the word.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @fherrerao 😃,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 4 above)
Comment (Whoistolu) [reply trimmer_decorator.rb]: All good now 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @fherrerao 

Great job implementing the recommended changes 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**